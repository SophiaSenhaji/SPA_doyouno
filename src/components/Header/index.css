.Header {
    display: flex;
    justify-content: space-between;
    margin-bottom: 8px;
}

.Header > * {
    display: flex;
    width: 30%;
}

.Header__title {
    justify-content: center;
}

.Header__right {
    justify-content: flex-start;
}

.Header__left {
    justify-content: flex-end;
}

.Header .Button, .Header a {
  font-weight: bold;
  cursor: pointer;
}