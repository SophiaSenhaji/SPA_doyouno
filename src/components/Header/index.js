import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './index.css';

const Header = ({ withSearch, withBack, withAdd, onSearchClick }) => (
    <div className="Header">
        {withSearch && <div className="Header__right Button" onClick={onSearchClick}>Search</div>}
        {withBack && <Link className="Header__right" to="/">{'<-'}</Link>}
        {!withSearch && !withBack && <div>{' '}</div>}
        <div className="Header__title">SPA</div>
        {withAdd && <Link className="Header__left" to="/form">+</Link>}
        {!withAdd && <div>{' '}</div>}
    </div>
)

export default Header;
