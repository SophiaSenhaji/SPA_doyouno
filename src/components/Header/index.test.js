import React from 'React';
import Header from './index.js';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Header component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<Header />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render correctly with withBack === true', () => {
        const wrapper = shallow(<Header withBack />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render correctly with withSearch === true', () => {
        const wrapper = shallow(<Header withSearch />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render correctly with withAdd === true', () => {
        const wrapper = shallow(<Header withAdd />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
})
