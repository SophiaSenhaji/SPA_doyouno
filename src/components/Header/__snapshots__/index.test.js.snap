// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div
  className="Header"
>
  <div>
     
  </div>
  <div
    className="Header__title"
  >
    SPA
  </div>
  <div>
     
  </div>
</div>
`;

exports[`Header component should render correctly with withAdd === true 1`] = `
<div
  className="Header"
>
  <div>
     
  </div>
  <div
    className="Header__title"
  >
    SPA
  </div>
  <Link
    className="Header__left"
    replace={false}
    to="/form"
  >
    +
  </Link>
</div>
`;

exports[`Header component should render correctly with withBack === true 1`] = `
<div
  className="Header"
>
  <Link
    className="Header__right"
    replace={false}
    to="/"
  >
    &lt;-
  </Link>
  <div
    className="Header__title"
  >
    SPA
  </div>
  <div>
     
  </div>
</div>
`;

exports[`Header component should render correctly with withSearch === true 1`] = `
<div
  className="Header"
>
  <div
    className="Header__right Button"
  >
    Search
  </div>
  <div
    className="Header__title"
  >
    SPA
  </div>
  <div>
     
  </div>
</div>
`;
